src/client/src/rt-components/timer/Timer.tsx

```diff
@@ -0,0 +1,36 @@
+import { PureComponent } from 'react'
+import { RequireOnlyOne } from 'rt-util'
+
+interface Props {
+  duration: number
+  immediate?: boolean
+  timeout: () => unknown
+  interval: () => unknown
+}
+
+export type TimerProps = RequireOnlyOne<Props, 'timeout' | 'interval'>
+
+export class Timer extends PureComponent<TimerProps> {
+  id: number
+
+  componentDidMount() {
+    const { interval, timeout, duration, immediate } = this.props
+
+    this.id = interval ? window.setInterval(interval, duration) : window.setTimeout(timeout, duration)
+
+    if (immediate) {
+      ;(interval || timeout)()
```

Comment:
Did you misplace the semi colon?

Reply:
not sure how that got in there 😳

Reply:
Apparently the `;` was added by prettier during a precommit task, and [this](https://github.com/prettier/prettier/issues/1244) seems to be the reason. 

There was a similar construct in the previous version of the code, I wondered the same when I first saw it.